### DISCUSSION TITLE
About partial_fit sgdclassifier

### DISCUSSION THREAD

**Discussion by SeymaYmn** [koaning/calmcode-feedback#215] [General]
Hi,
I saw your website about partial_fit sgdclassifier. Thank you for the information you provided. I want to train a data set containing 4500 files with SVM without loading it into RAM. I think the partial_fit method is suitable for this. But I'm having trouble. how can I arrange test_set, train_set? I couldn't figure out how to set these. Can you help me?

**Comment by koaning** [comment]
Couldn't you just run `train_test_split` on every file? The train part can then be used to train the SVM and once everything is trained you might move on to repeat the exercise but use the test sets for evaluation?

**Reply by SeymaYmn** [reply]
So, do you suggest that I remove my test set from the entire set and train the remaining set, and then use the test set according to the resulting model? 
And also I divide my data set into 400 data sets. I read every 400 files, preprocess them and combine them. I use to train partial_fit after every 400 files. I think it learns incrementally after 400 files from each cycle. Is it true? Or should I follow another step after each for loop? I'm adding the learning part of my code below. When I train this way, my confusion matrix right column turns out to be 0.
#my python code
    svm_model = SGDClassifier(loss='hinge', shuffle=True)    
    batch_size = 400
    for i in range(0, len(filelabels), batch_size):
        file_group = filelabels.iloc[i:i+batch_size]
        alldataList = []
        for index, row in file_group.iterrows():
            data = pd.read_parquet("../data/parquet/train/"+row['filepath'], engine="pyarrow")
            ...
            data = preporcessing(data)
            ...
            alldataList.append(data)          
        alldata = pd.concat((alldataList), sort=False)
        X = alldata.drop(['label'], axis=1)
        y = alldata['label']
        del alldata
        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
        svm_model.partial_fit(X_train, y_train, classes=np.unique(y))
